WordPress/gutenberg
# Rename "auto inserting blocks" to "block hooks"

## What?
Rename "auto inserting blocks" to "block hooks".

Additionally, this PR contains two minor, unrelated changes:
- [Fix a REST API schema description](https://github.com/WordPress/gutenberg/pull/54147/files#r1315078729).
- Remove a TODO note that's obsolete (as of #54029): https://github.com/WordPress/gutenberg/pull/54147/commits/8a5ed7aa0223d434242d532c3d37f4253e4ee782

## Why?
See https://github.com/WordPress/gutenberg/issues/53987#issuecomment-1695915874:

> I've seen anecdotal feedback that `autoInsert` is not the clearest of descriptions. I'd like to propose renaming to the more familiar `hooks` terminology—and "block hooks" in more general terms—to help folks understand the mechanics and purpose more rapidly.

## How?
Replacing strings, of course. Furthermore, by hopefully picking the right replacements for different nouns and verbs previously used for auto-inserting blocks with their block hook counterpart.

For one, while [_hooks_](https://developer.wordpress.org/plugins/hooks/) is a familiar concept in WordPress, it's not really used in Core's functions names (we have `add_filter` and `add_action` but not `add_hook`); we might want to consider this for our block hook related function names.

For example, should `gutenberg_register_auto_inserted_block` become `gutenberg_add_block_hook`? That seems misleading, like it would actually add the block _hook_ -- i.e. the location where it is inserted -- rather than the block. So maybe just `gutenberg_add_block`? That will become `add_block_hook` in Core 😕 I ended up using `gutenberg_register_hooked_block` in this case.

Or: in `block.json`, should the field be called `blockHooks` or just `hooks` -- since implicitly, `block.json` is always about blocks?

## Testing Instructions
**Note that you have to re-enable the experiment, as its name has changed as well!**

You can use [v0.3.1](https://github.com/ockham/like-button/releases/download/v0.3.1/like-button.zip) of the Like Button block plugin for testing (it's been updated to use `__experimentalBlockHooks` instead of `__experimentalAutoInsert`).

Verify that ~auto-inserting blocks~ block hooks work as before.
